src/modules/stakewise/stakewise-daemon/common/contracts/split-main.go

```diff
@@ -0,0 +1,67 @@
+)
+
+// ABI cache
+var splitMainAbi abi.ABI
+var splitMainOnce sync.Once
+
+// Binding for Splitmain
+type SplitMain struct {
+	contract *eth.Contract
+	txMgr    *eth.TransactionManager
+}
+
+// Create a new Splitmain instance
+func NewSplitMain(address common.Address, ec eth.IExecutionClient, txMgr *eth.TransactionManager) (*SplitMain, error) {
+	// Parse the ABI
+	var err error
+	splitMainOnce.Do(func() {
+		var parsedAbi abi.ABI
+		parsedAbi, err = abi.JSON(strings.NewReader(splitMainAbiString))
+		if err == nil {
+			splitMainAbi = parsedAbi
+		}
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error parsing Stakewise Vault ABI: %w", err)
+	}
+
+	// Create the contract
+	contract := &eth.Contract{
+		ContractImpl: bind.NewBoundContract(address, splitMainAbi, ec, ec, ec),
+		Address:      address,
+		ABI:          &splitMainAbi,
+	}
+
+	return &SplitMain{
+		contract: contract,
+		txMgr:    txMgr,
+	}, nil
+}
+
+// =============
+// === Calls ===
+// =============
+
+// ====================
+// === Transactions ===
+// ====================
+
+func (c *SplitMain) SetWithdraw(address common.Address, opts *bind.TransactOpts) (*eth.TransactionInfo, error) {
```

Comment:
This should just be called `Withdraw` since that's what the contract method's name is. Also I assume the other arguments (like `big.NewInt(0)` and `[]common.Address{}`) are placeholder and that's what you were talking about earlier today with the unified contract? If so you can just leave them here for now and we'll adjust later. Otherwise they have to be populated from *somewhere* and passed in to this function.

Reply:
No, these are different parameters. The values are ethAmount (0 = withdraw all) and list of erc20 tokens to withdraw from (empty for now). I'll update this

Reply:
I moved these values into resources.go